.github/workflows/build.yml

```diff
@@ -0,0 +1,23 @@
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+name: Build
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
```

Comment:
If we pin, I suggest we pin straight to git hashes, as dependabot / renovate can still upgrade these, but they also give us the added benefit of being unable to change.

Updated (at least via Renovate, haven't used Dependabot in a while) would look like this: https://github.com/FoxDenHome/LCDify/pull/20/files

Reply:
> but they also give us the added benefit of being unable to change.

Sorry, I'm afraid I didn't get this. The versions would have their own benefit of being easier to track for specific changes directly in the release notes, instead of having to seek for the matching commit hashes. I might be missing something of course.

Reply:
I was thinking since git hashes are immutable, but git tags are only suggested to be immutable. But yeah, probably not worth it to sacrifice readability